### ISSUE TITLE
gradient needs to be divided by batch size v2

### ISSUE THREAD

**Issue by zchky** [jxgu1016/MNIST_center_loss_pytorch#10]
https://github.com/jxgu1016/MNIST_center_loss_pytorch/issues/9
this issue has been raised before.

but I find the average operation only influence the gradient of Xi, which the gradient of centers didn't be divided by batch size.

for example:
if the gradient of Xi is Grad_Xi, and gradient of Center is Grad_center when loss = LO
if LO/10, the gradient of Xi will be Grad_Xi/10, but the gradient of Center still be Grad_center

Could you help to check it?

**Comment by jxgu1016** [comment]
That's correct that the grads of the centers won't be averaged. And I am not sure whether we should do the average processing, so I left it not.

**Comment by zchky** [comment]
@jxgu1016 thanks for your quick reply

i don't understand why you only average the gradient of Xi but leave Centers alone.

i think that in the normal situation, we average the total loss which means all the gradient will be divided by batch size.

**Comment by jxgu1016** [comment]
@zchky Could you check the latest version in the pull request?
